### ISSUE TITLE
parameter 'algorithm' from BSS mask the 'algorithm' argument from sklearn.fastica

### ISSUE THREAD

**Issue by CristianNecula** [hyperspy/hyperspy#1678]
In the blind_source_separation function they are two 'algorithm' parameters. One belongs to the main blind_source_separation function (change between different types of ICA, e.g. between fastica, jade, cubica etc.). The other one belongs to the sklearn.decomposition.fastica function and allows to change between 'parallel' and 'deflation' options in the FastICA. The two arguments  are masking each other.

**Comment by tjof2** [comment]
How about `method` for `[fastica, jade, cubica]` and `algorithm` to go to sklearn?

**Comment by ericpre** [comment]
xref https://github.com/hyperspy/hyperspy/pull/2035#discussion_r388253585 on making arguments more consistent between `blind_source_separation` and `decomposition`.
